File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3095_shortest-subarray-with-or-at-least-k-i/solutions/python3/solution.py
from typing import List

class Solution:
    def minimumSubarrayLength(self, nums: List[int], k: int) -> int:
        n = len(nums)
        min_length = float('inf')
        
        for start in range(n):
            or_value = 0
            for end in range(start, n):
                or_value |= nums[end]
                if or_value >= k:
                    min_length = min(min_length, end - start + 1)
                    break
        
        return min_length if min_length != float('inf') else -1

File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3095_shortest-subarray-with-or-at-least-k-i/solutions/cpp/solution.cpp
#include <vector>
#include <algorithm>

class Solution {
public:
    int minimumSubarrayLength(std::vector<int>& nums, int k) {
        int n = nums.size();
        int min_length = n + 1; // Start with a non-possible maximum length
        for (int start = 0; start < n; ++start) {
            int current_or = 0;
            for (int end = start; end < n; ++end) {
                current_or |= nums[end];
                if (current_or >= k) {
                    min_length = std::min(min_length, end - start + 1);
                    break; // Found the shortest subarray starting at `start`
                }
            }
        }
        return min_length <= n ? min_length : -1;
    }
};
